### DISCUSSION TITLE
自动化辅助增强

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3596] [插件说明]
描述
---

> 增强简悦的自动化服务，提供更多监控模块。

地址
---

[https://simpread.ksria.cn/plugins/details/DH9l5jblPH](https://simpread.ksria.cn/plugins/details/DH9l5jblPH)

监控对象
---

**简悦 2.2.0 版**的自动化方案，只能监控：

- 加入稍后读

- 新建、更改、删除标注

这两个场景，此插件增加了新的监控模块：

- 稍后读标签变化

- 稍后读内容改变（监控稍后读编辑框的保存，包括：标题、描述、备注、外部引用）

- 标注的内容改变（监标注读编辑框的保存，包括：备注、外部引用）

以及

- 加入稍后读时对快捷键的支持

- 加入标注时（包括：删除和标注色的改变）对快捷键的支持

此方式可自动执行（支持快捷键的）简悦插件

至此，自动化辅助增强基本上涵盖了简悦在稍后读、标注时的全部变化（增 / 删 / 改）

使用场景
---

> 以下场景都是简悦自动化没法实现，但通过此插件可以实现的功能：

1. 加入稍后读 **自动导出到 Obsidian**

2. 对加入的稍后读更改标签时 **自动调用 [Notion 辅助增强](https://github.com/Kenshin/simpread/discussions/3572)** 更新 Notion Page

3. 对标注的标签、备注修改时 **自动调用设置好的自动化方案**，如：导出 HTML

4. 对稍后读备注、外链修改时 **自动触发 XXXX**

5. 修改稍后读的标题后 **自动触发 XXXX**

更多使用场景例子 [请看这里](https://github.com/Kenshin/simpread/discussions/3596#discussioncomment-2441614)。

选项
---

> 随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角触发器 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

![image](https://user-images.githubusercontent.com/81074/161675185-57edd5e0-a94c-440c-967d-59c76ce39413.png)

功能说明
---

### 1. 稍后读的监控

![image](https://user-images.githubusercontent.com/81074/161695424-01ff74bf-eb06-42c1-80a7-87f87077ebed.png)

上面编号 ➊ ~ ➍ 对应的功能分别是：

1. 来自选项页 → 自动化的设置项，为了方便用户查看，无法再此界面编辑。

2. 当加入稍后读后，执行自动触发的快捷键，如：`o b` → 自动导出到 Obsidain `e n t`  → 导入到 Notion 辅助增强

3.  当改变稍后读的元数据，如：备注、外部链接、标签时，执行自动触发的快捷键

4. 当改变稍后读的元数据，如：备注、外部链接、标签时，执行自动触发的服务

### 2. 标注的监控

![image](https://user-images.githubusercontent.com/81074/161694961-b86db4af-5852-4d30-a138-2a9c1801e435.png)

上面编号 ➊ ~ ➍ 对应的功能分别是：

1. 来自选项页 → 自动化的设置项，为了方便用户查看，无法再此界面编辑。

2. 当加入标注后，执行自动触发的快捷键，如：`o b` → 自动导出到 Obsidain `e n t`  → 导入到 Notion 辅助增强

3.  当改变标注的元数据，如：备注、外部链接、标签时，执行自动触发的快捷键

4. 当改变标注的元数据，如：备注、外部链接、标签时，执行自动触发的服务

### 3. 优化提示信息

<img width="575" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/5902e5d6-e4a7-468e-bcdf-237db30c8f25">

开启后，仅在标注并触发自动化时生效，会优化：`已成功保存。` `正在保存` `已成功保存到XXX` `正在导入` 等反复出现的提示，[详细说明](https://github.com/Kenshin/simpread/discussions/3596#discussioncomment-6687745)。

权限
---

插件是基础账户就能使用的功能，但此插件为 [自动化](http://ksria.com/simpread/docs/#/%E8%87%AA%E5%8A%A8%E5%8C%96) 功能的辅助增强插件，**自动化功能为高级账户功能**，所以此插件 **需要高级账户权限才能使用**，否则会出现下图提示。

![image](https://user-images.githubusercontent.com/81074/161732918-a2aa5c8a-1780-4689-9377-1c61efd54f9a.png)

注意
---

因 1.1.0 版数据结构调整，如果你曾经使用过此插件，更新到 1.1.0 版后会重新让你设置自动化方案。

与扩展端自动化方案区别
---

1. 扩展端自动化方案没有此插件 [监控对象](https://github.com/Kenshin/simpread/discussions/3596#discussioncomment-2441817) 多。

2. 此插件只能运行在阅读模式中，暂时无法运行在稍后读中，而扩展端则可以运行在稍后读中。

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过 [此链接](https://simpread.ksria.cn/plugins/details/DH9l5jblPH) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

更新日志
---

### 1.2.3

- `Add` 选项卡左下角版本号（点击可验证是否有新版）

### 1.2.2

- `Add` 顶级参数 `SRautomatepause`

- `Add` 自动化辅助增强插件选项页左下角版本号

### 1.2.1

- `Add` 顶级函数 `SRenhanautomate` 传入参数包括：`unread_add` `unread_changed` `annote_add` `annote_changed` 

### 1.2.0

- `Add` 在非稍后读状态使用鼠标划选加入稍后读时的监控。

### 1.1.1

- `Add` [优化提示信息](https://github.com/Kenshin/simpread/discussions/3596#discussioncomment-6687745)

### 1.1.0

重构了整个自动化增强的流程，将 **稍后读** 和 **标注** 的自动化监控分开，因此现在每种（稍后读 / 标注）都具有如下的功能：

- 稍后读

   1. 添加 → 自动触发导出服务和快捷键

   3. 元数据的改变 → 自动触发导出服务和快捷键

- 标注

   1. 添加（包括：删除和更改标注色） → 自动触发导出服务和快捷键

   2. 元数据的改变 → 自动触发导出服务和快捷键

### 1.0.1

- `Add` 当 `添加` 或 `删除` 标注、改变标注颜色时触发自动化

### 1.0.0

- 自动监控

   - 稍后读标签变化

   - 稍后读内容改变（监控稍后读编辑框的保存）

   - 标注编内容改变（监控标注编辑框的保存）

   - 加入稍后读的监控（支持触发快捷键）

- 自动触发

   - 导出功能

   - 快捷键

**Comment by Kenshin** [comment]
# 快捷键支持

简悦大部分功能都支持快捷键操作，但此输入框建议仅输入插件的快捷键，例如：

- `o b` → [导入到 Obsidain](https://github.com/Kenshin/simpread/discussions/2902) 的快捷键

- `e n t` → [导入到 Notion 辅助增强](https://github.com/Kenshin/simpread/discussions/3572) 的快捷键

# 一个简单用法

下图的自动化是：当加入稍后读后，自动导入到 Notion、HTML、Pocket 与 Obsdian

![image](https://user-images.githubusercontent.com/81074/161676953-e9f6e62f-bc46-4ba8-90b4-f1f40966c826.png)

# 如何获得插件的快捷键

找到插件详细页例如 [导入到 Obsidain](https://simpread.ksria.cn/plugins/details/1VQ19jCD8Z)，然后就能得知这个插件的快捷键。

![image](https://user-images.githubusercontent.com/81074/160228112-8dbc4141-ea90-4df3-b241-5fb833ce7452.png)

**Reply by T0TT** [reply]
你好，关于 自动化辅助增强插件 目前和 离线下载辅助增强插件 快捷键之间好像有bug，我尝试在 阅读模式 下用DD和右键都不能触发 离线下载辅助增强插件。但是在 稍后读的自动化增强 板块放m t 快捷键（将标注以 MD 2.0 模板形式导出）以同样方式操作是可用的。
我使用 这两个插件结合 主要是为了在下载时防止有些网站如 少数派 有图片防盗链的即便在 扩展端 使用 离线HTML 下载也无法下载图片，不如什么网站也一样直接全部用离线HTML更保险。
<img width="629" height="710" alt="{3E3BB5C7-F585-4B81-97B4-BCC7E7B4CCAC}" src="https://github.com/user-attachments/assets/cb65e09f-299d-4bbc-ab07-19cbc1cc4bb9" />

**Comment by Kenshin** [comment]
# 使用场景

## 加入稍后读自动导入到 Notion，包括更新稍后读标签后也同步到 Notion

1. 设置加入到稍后读时，触发 **自动导入到 Notion**

2. 设置改变稍后读元数据时，触发 **导入到 Notion 辅助增强** 的快捷键

3. 设置加入标注时，触发 **导入到 Notion 辅助增强** 的添加标注快捷键

4. 设置改变标注元数据时，触发 **导入到 Notion 辅助增强** 的添加标注快捷键

![image](https://user-images.githubusercontent.com/81074/161678385-7719a746-78f8-4362-be43-d4d6764cb931.png)

## 加入稍后读自动导入到 Obsidian
  
![image](https://user-images.githubusercontent.com/81074/161678691-5d6d02cc-3d52-4512-ac06-11cb1ff05297.png)

## 加入稍后读自动保存 HTML，标注更改时自动导出 HTML

<img width="1063" alt="2022-04-05_12-33-10" src="https://user-images.githubusercontent.com/81074/161679733-7aa04e43-1baf-4bf7-88aa-382fa4fb61b1.png">

## 利用此功能改善离线 HTML 的自动化体验

如果你将标注的自动化设置为 **离线 HTML**，每增加一个标注都会触发一次保存离线 HTML的工作，这样非常消耗性能，你可以将此流程修改为：

1. 不设置标注的自动化
2. 加入稍后读和稍后读的元数据化时，导出离线 HTML

<img width="1128" alt="2022-04-05_12-40-47" src="https://user-images.githubusercontent.com/81074/161680153-5b6c1d00-df7b-4a08-9fed-3d53cc885deb.png">

**Comment by Kenshin** [comment]
# 监控对象

**简悦 2.2.0 版**的自动化方案，只能监控：

- 加入稍后读
- 新建、更改、删除标注

这两个场景，此插件增加了新的监控模块：

- 稍后读标签变化
- 稍后读内容改变（监控稍后读编辑框的保存，包括：标题、描述、备注、外部引用）
- 标注的内容改变（监标注读编辑框的保存，包括：备注、外部引用）

以及

- 加入稍后读时对快捷键的支持
- 加入标注时（包括：删除和标注色的改变）对快捷键的支持

此方式可自动执行（支持快捷键的）简悦插件

至此，自动化辅助增强基本上涵盖了简悦在稍后读、标注时的全部变化（增 / 删 / 改）

**Reply by 1055373165** [reply]
<img width="861" alt="image" src="https://user-images.githubusercontent.com/33158355/184487166-47deda30-3add-4cc8-802c-7955855d399d.png">
<img width="514" alt="image" src="https://user-images.githubusercontent.com/33158355/184487463-5de97502-2d72-4105-a6d2-efc4b02a0a5f.png">
这种情况怎么处理？

讨论转移到 https://github.com/Kenshin/simpread/issues/4375

**Reply by TianMaXingKong2003** [reply]
我按照您的方式，设置了b b b来将标注自动导入到obsidian中，请问右上角总是常驻“已保存到obsidian，是否打开？”这个通知要如何才能不让它常驻或者不再出现呢？

**Reply by Kenshin** [reply]
@TianMaXingKong2003 

暂时没有，但这个功能早就已经在 Todo 里面了。😄

**Comment by Kenshin** [comment]
# 优化提示信息

<img width="575" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/5902e5d6-e4a7-468e-bcdf-237db30c8f25">

## 优化内容

开启后，仅在标注并触发自动化时生效，会优化：`已成功保存。` `正在保存` `已成功保存到XXX` `正在导入` 等反复出现的提示。

## 注意

极端情况下，如：安装了太多的插件并使用同步助手的自动同步以及连续标注时，会在浮动工具条显示时稍有卡顿问题。

注意：仅为个例，在测试期间除了我，几乎没有用户反馈过类似问题。
